src/python/pants/backend/python/goals/lockfile.py

```diff
@@ -156,20 +156,21 @@ def warn_python_repos(option: str) -> None:
 
 @dataclass(frozen=True)
 class _PipArgsAndConstraintsSetup:
+    resolve_config: ResolvePexConfig
     args: tuple[str, ...]
     digest: Digest
-    constraints: FrozenOrderedSet[PipRequirement]
-    only_binary: tuple[str, ...]
-    no_binary: tuple[str, ...]
 
 
 @rule_helper
-async def _setup_pip_args_and_constraints_file(resolve_name: str) -> _PipArgsAndConstraintsSetup:
-    args = []
-    digests = []
+async def _setup_pip_args_and_constraints_file(
+    resolve_name: str, *, use_pex: bool
+) -> _PipArgsAndConstraintsSetup:
     resolve_config = await Get(ResolvePexConfig, ResolvePexConfigRequest(resolve_name))
 
-    if resolve_config.no_binary or resolve_config.only_binary:
+    args = list(resolve_config.indexes_and_find_links_and_manylinux_pex_args())
```

Comment:
Since this is the only `use_pex` oblivious thing accomplished by this `@_rule_helper`, maybe it should just be called outside to avoid that parameter...?

Reply:
I'm not sure I follow. We need the `_PipArgsAndConstraintsSetup.resolve_config` either way at the bottom of this rule to be able to set the lockfile metadata, even if it's a Poetry lockfile.